packages/application-components/src/experimental/components/internals/option/option.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { NO_VALUE_FALLBACK } from '@commercetools-frontend/constants';
+import Spacings from '@commercetools-uikit/spacings';
+import Text from '@commercetools-uikit/text';
+import AsyncSelectInput from '@commercetools-uikit/async-select-input';
+
+type TData = {
+  key: string;
+  label: string;
+  value: string;
+};
+
+type TProps = {
+  data: TData;
+};
+
+const Option = <T extends TProps>(props: T): JSX.Element => {
```

Comment:
Asking just out of personal curiosity: why not just 

```
const (props: TProps): JSX.Element => {
```

TS will complain that additional props are passed in this case?

Reply:
hey, thanks for this question.
What I want to achieve here is that the consumer can pass more props than `props.data`.

as you can see in `L19`, we spread the props to the underlying component.
If the consumer implements a select option that consumes this, they can define their own _required_ props top of `TProps`.

At least that was my aim.

Reply:
React.ReactNode and JSX.Element always confused me as to when to use which. So I'll just share that here too. A ReactNode is a ReactElement, a ReactFragment, a string, a number or an array of ReactNodes, or null, or undefined, or a boolean. A SX.Element is a ReactElement, with the generic type for props and type being any.

This always drove me nuts. I kept the rule of thumb that a ReactNode is what render on a class component returns and is used if you e.g. allow to return a string A JSX.Element is only for Functional Components.

That said: if an option can be a string or undefined then this might want to be ReactNode.

Reply:
A `JSX.Element` is what usually TS uses/infers when you don't explicitly define the return type. Which is also how  `FC` is declared.

<img width="795" alt="image" src="https://user-images.githubusercontent.com/1110551/102186443-ec084d80-3eb2-11eb-9a4e-edd4318079d4.png">

So it's fine here. A `ReactNode` is what you use for example for typing `children`.

Reply:
> If the consumer implements a select option that consumes this, they can define their own required props top of TProps.

This component is only used internally, so we don't really need to make it unnecessarily generic.
You can keep it simple maybe.
I'm also fine either way.

Reply:
What I said: JSX.Element is for React.FC. ReactNode for ClassComponent.

Reply:
You can always "force it" to return a `JSX.Element` by wrapping it into a fragment, for example:

```jsx
const Text = () => <>Hello</>
```